banking/klarna_kosma_integration/doctype/bank_reconciliation_tool_beta/bank_reconciliation_tool_beta.py

```diff
@@ -616,14 +618,18 @@ def get_ld_matching_query(bank_account, exact_match, filters):
 		"party_type"
 	) and loan_disbursement.applicant == filters.get("party")
 
-	rank = frappe.qb.terms.Case().when(matching_reference, 1).else_(0)
+	date_condition = Coalesce(
+		loan_disbursement.reference_date, loan_disbursement.disbursement_date
+	) == Parameter("%(date)s")
+	date_rank = frappe.qb.terms.Case().when(date_condition, 1).else_(0)
 
+	rank = frappe.qb.terms.Case().when(matching_reference, 1).else_(0)
 	rank1 = frappe.qb.terms.Case().when(matching_party, 1).else_(0)
```

Comment:
```suggestion
	reference_rank = frappe.qb.terms.Case().when(matching_reference, 1).else_(0)
	party_rank = frappe.qb.terms.Case().when(matching_party, 1).else_(0)
```

Maybe we can use this opportunity for a minor refactor